src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -295,6 +295,9 @@ void remove_redundant_reorders::run(program& p) {
         auto o_layout = r_node.get_output_layout();
         const auto& i_layout = r_node.get_input_layout(0);
 
+        if (r_node.get_dependency(0).is_type<crop>())
```

Comment:
Do we skip reorder optimizing which has crop dep even it does not has any padding?

Reply:
Yes. reorder opt is determined in remove_redundant_reorders and crop padding is determined in prepare_buffer_fusing. And remove_redundant_reorders is executed before prepare_buffer_fusing. So padding info isn't known in remove_redundant_reorders pass.

Reply:
`if (!update_implementations || (r_node.get_dependency(0).is_type<crop>() && r_node.get_dependency(0).can_be_optimized())`

Reply:
At post optimization graph phase, update_implementations is true. 
At this phase, buffer fusing is already applied, so we can check the optimization of crop.

Reply:
As update_implementations needed to be confined to crop only because the code below the condition should be run sometimes, the condition was updated.